plugins/woocommerce/src/Internal/Admin/Notes/WooSubscriptionsNotes.php

```diff
@@ -274,27 +265,21 @@ public function add_or_update_subscription_expiring( $subscription ) {
 
 		$note = $this->find_note_for_product_id( $product_id );
 
-		if ( $note ) {
-			$content_data = $note->get_content_data();
-			if ( property_exists( $content_data, 'days_until_expiration' ) ) {
-				// Note: There is no reason this property should not exist. This is just defensive programming.
-				$note_days_until_expiration = intval( $content_data->days_until_expiration );
-				if ( $days_until_expiration === $note_days_until_expiration ) {
-					// Note is already up to date. Bail.
-					return;
-				}
+		// Note: There is no reason this property should not exist. This is just defensive programming.
+		if ( $note && property_exists( $note->get_content_data(), 'days_until_expiration' ) ) {
+			$note_days_until_expiration = intval( $note->get_content_data()->days_until_expiration );
+			if ( $days_until_expiration === $note_days_until_expiration ) {
+				// Note is already up to date. Bail.
+				return;
+			}
 
-				// If we have a note and we are at or have crossed a threshold, we should delete
-				// the old note and create a new one, thereby "bumping" the note to the top of the inbox.
-				$bump_thresholds    = $this->get_bump_thresholds();
-				$crossing_threshold = false;
-
-				foreach ( (array) $bump_thresholds as $bump_threshold ) {
-					if ( ( $note_days_until_expiration > $bump_threshold ) && ( $days_until_expiration <= $bump_threshold ) ) {
-						$note->delete();
-						$note = false;
-						continue;
-					}
+			// If we have a note and we are at or have crossed a threshold, we should delete
+			// the old note and create a new one, thereby "bumping" the note to the top of the inbox.
+			foreach ( (array) self::BUMP_THRESHOLDS as $bump_threshold ) {
+				if ( ( $note_days_until_expiration > $bump_threshold ) && ( $days_until_expiration <= $bump_threshold ) ) {
+					$note->delete();
+					$note = false;
+					break;
```

Comment:
I guess this is the critical difference. I'll have to have another look at this tomorrow to try and understand what we're doing here.

* `$note_days_until_expiration` is days until the note expires, that is, until we want to stop showing it.
* `$days_until_expiration` is days until the subscription expires.
* We loop through the array of bump thresholds, which is ordered largest to smallest.
* If the note is due to expire after a threshold and the subscription to expire sooner than the threshold, delete the note and break from the loop.

I guess I need to understand whether we omit any useful thresholds by breaking here.

Reply:
@andfinally when we delete the `$note` it's already false, so we can't delete it again :)

This is why it only makes sense to do it once, and break. The thresholds are only used to check whether deletion should happen or not

Reply:
I get it now, thanks. There is only one `$note`, so once deleted we don't want to do anything else.